File: RecyclerView/Java/app/src/main/java/com/example/javarecyclerexample/AnimeAdapter.java
package com.example.javarecyclerexample;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class AnimeAdapter extends RecyclerView.Adapter<AnimeAdapter.AnimeViewHolder>{

    List<Anime> animeList;

    public AnimeAdapter(List<Anime> animeList) {
        this.animeList = animeList;
    }

    @NonNull
    @Override
    public AnimeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_line, parent, false);
        return new AnimeViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull AnimeViewHolder holder, int position) {

        Anime anime = animeList.get(position);

        holder.nome.setText(anime.getNome());
        holder.protagonista.setText(anime.getProtagonista());

    }

    @Override
    public int getItemCount() {
        return animeList.size();
    }

    public class AnimeViewHolder extends RecyclerView.ViewHolder {

        TextView nome;
        TextView protagonista;

        public AnimeViewHolder(@NonNull View itemView) {
            super(itemView);

            nome = (TextView)itemView.findViewById(R.id.nomeAnime);
            protagonista = (TextView)itemView.findViewById(R.id.nomeProtagonista);

        }
    }
}

File: RecyclerView/Kotlin/KotlinRecyclerExample/app/src/main/java/com/example/kotlinrecyclerexample/AnimeAdapter.kt
package com.example.kotlinrecyclerexample

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.recycler_line.view.*

class AnimeAdapter(val animeList: List<Anime>) : RecyclerView.Adapter<AnimeAdapter.AnimeViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AnimeViewHolder {

        var itemView : View = LayoutInflater.from(parent.context)
            .inflate(R.layout.recycler_line, parent, false)

        return AnimeViewHolder(itemView)

    }

    override fun getItemCount(): Int {
        return animeList.size
    }

    override fun onBindViewHolder(holder: AnimeViewHolder, position: Int) {

        val anime = animeList.get(position)

        holder.nome.setText(anime.nome)
        holder.protagonista.setText(anime.protagonista)

    }

    class AnimeViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){

        val nome = itemView.nomeAnime
        val protagonista = itemView.nomeProtagonista

    }
}